### ISSUE TITLE
No loop closure detected

### ISSUE THREAD

**Issue by YilmazKadir** [MIT-SPARK/Kimera-VIO#72]
I have looked at all issues but I could not find a related one. When I run with loop closure detection enabled, it does not detect any loops. I uploaded the output_lcd_status.csv file. I do not know if I am doing something wrong. Can you please help me?

Command: bash ./scripts/stereoVIOEuroc.bash -p ~/EuRoC/V1_01_easy/ -lcd -log

[output_lcd_status.csv.zip](https://github.com/MIT-SPARK/Kimera-VIO/files/4624124/output_lcd_status.csv.zip)

**Comment by ToniRV** [comment]
Hi @kdr193,
Alpha is what controls the number of loop closures detected, try to increase it, as it is now low: `alpha: 0.001`
Check in [here](https://github.com/MIT-SPARK/Kimera-VIO/blob/master/params/Euroc/LcdParams.yaml).

**Comment by WouterJansen** [comment]
I'm experiencing the same issue. Changed the alpha value to 0.1 but still no LC detected;

**Comment by WouterJansen** [comment]
@ToniRV 

I'm continuing to experience many issues regarding loop closures and (R)PGO in the latest version. In the V1 euroc set I have to set the LCD parameters to
```
alpha: 1
min_temporal_matches: 0
```
to detect a few closures. And I don't believe they are in any way correct.

I know this is in the wrong repository to say but on top of that none of the LC/RPGO related ROS topics does anything get published. 
```
/kimera_vio_ros/graph_nodes
/kimera_vio_ros/loop_edges
/kimera_vio_ros/optimized_trajectory
/kimera_vio_ros/pose_graph
/kimera_vio_ros/rejected_loop_edges
```
All those topics are empty. So it's also rather difficult to test the system this way.

Is this something you guys are experiencing yourselves or does this continue to just be an issue for certain users? I'm using the current master updated and nothing changed except LCD turned on (and those LCD 2 parameters tweaked as stated above).

**Comment by WouterJansen** [comment]
As stated above, currently can get a few loop closures in the euroc dataset if I set `alpha` to 0.001 and `min_temporal_matches`  to 0. However, I am not certain these are detected correctly now or are false positives. Hard to tell in Euroc where there should be closures detected just visually.

In my own dataset I also get similar results but granted it is not the best dataset (taken in simulation, low rates of IMU and camera) but it does not a bad job at the state estimation with Kimera-VIO. However, the LCD did not detect some very obvious loops. The biggest reason seems to be _LOW_NSS_FACTOR_. I saw this meant normalized similarity score but didn't really understand what it meant in terms of LCD specifically in Kimera. Judging by the code it analyzes mostly if you are standing still by comparing current to previous DBOW vector? I might be interpreting that wrong. However, when turning it off (`use_nss: 0`) for testing the obvious loop point was detected correctly. 

Just thought I'd share this with others having these issues with their datasets
